### DISCUSSION TITLE
Interact with eloquent in vendor package

### DISCUSSION THREAD

**Discussion by bechir** [laravel/framework#44694] [Q&A] [answered]
Hello everyone,

Is there a way to query on database inside a vendor package using Reflection API?
For example I want to fetch all entries of a model from database:

```php
// VendorPackage/Foo.php

class Foo {
    // $model is a Model class in App\\Models\\ namespace
    public function bar(string $model)
    {
        $reflection = new \ReflectionClass($model);
        \call_user_func(sprintf('%s::all', $reflectionClass->getName())
    }
```

But actually this is not working because the connection is null at this point.
`In Model.php line 1811: Call to a member function connection() on null `

I want to access to the model only after attributes (like the table name) are filled.

**Comment by s-patompong** [comment]
I’m not sure about the reflection approach either. Do you have a reason why you don’t just send the model object to this method?

**Comment by henzeb** [comment, marked as answer]
I'm not even sure what you are trying to do.  Are you using Orchestra testbench in your tests?

besides, you don't need the Reflection API. You can just call `$model::all();` after checking if the string is a model using is_subclass_of

**Comment by bechir** [comment]
Thank you.
@henzeb `$model::all()` is working well after verifying inheritance with `is_subclass_of`.
